### PR TITLE
continuously update queue

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
Initial tests seem to make `gimme scan` +/- 25% faster by updating the queue online. It **seems** to not have an effect on memory usage. 

new implementtation:

```
	User time (seconds): 10842.61
	System time (seconds): 89.40
	Percent of CPU this job got: 935%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 19:29.19
	Maximum resident set size (kbytes): 760520
```

vs

old implementation:

```
	User time (seconds): 10953.35
	System time (seconds): 101.56
	Percent of CPU this job got: 693%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 26:33.91
	Maximum resident set size (kbytes): 775536
```

According to `line_profiler` the actual `_scan_jobs` function takes +/- 65% of the time of the previous implementation.

@simonvh Does this get tested by gimmemotifs CI? I did a quick scan by checking if the shasum was the same with the old and new method, but I don't feel confident that it works with only that as a test :smile:

### CODE REVIEW

**Comment by simonvh** [comment]
It should get tested by the CI, yes.

File: gimmemotifs/scanner.py [hunk 1]
```diff
@@ -1164,20 +1166,31 @@ def _scan_jobs(self, scan_func, scan_seqs):
...
+                jobs.append(job)
+
+                while (len(jobs) > max_queue_size) and not jobs[0].ready():
+                    time.sleep(0.05)
+                    if jobs[0].ready():
+                        for ret in jobs[0].get():
+                            region = scan_seqs[k]
+                            k += 1
+                            yield region, ret
+                    jobs = jobs[1:]
```
Comment (siebrenf) [inline gimmemotifs/scanner.py:1186]: shouldn't this line be inside the if-statement?

**Comment by Maarten-vd-Sande** [comment]
CI doesn't actually check if it works. It just checks if the command can be run I believe


Small tests (ignore the rude naming):
```
(gimme) sande@ocimum:/vol/gimme$ wc -l newandcool.tsv oldandboring.tsv 
   122853 newandcool.tsv
   122853 oldandboring.tsv
(gimme) sande@ocimum:/vol/gimme$ shasum newandcool.tsv oldandboring.tsv 
99130338bed9bb3e968c2c43ed47806efd4f6af7  newandcool.tsv
99130338bed9bb3e968c2c43ed47806efd4f6af7  oldandboring.tsv
(gimme) sande@ocimum:/vol/gimme$ md5sum newandcool.tsv oldandboring.tsv 
b40fb6fcf444048bb4afe201bbcef07c  newandcool.tsv
b40fb6fcf444048bb4afe201bbcef07c  oldandboring.tsv

```

```
>>> old = pd.read_table("oldandboring.tsv", comment="#")
>>> new = pd.read_table("newandcool.tsv", comment="#")
>>> (old == new).all().all()
True
```

However bad-ish news, this is only +/- 10% faster. I previously tested in on `cn45` and I am not sure if the increase in performance there was bigger, or I was running it on a faulty version of the patch...

**Comment by simonvh** [comment]
Well, faster is faster, even if it's 10%.
There should be checks to test the scanning functionality, I believe? But maybe I'm mistaken.

**Comment by Maarten-vd-Sande** [comment]
My bad, it does get tested :innocent:

**Review by simonvh — APPROVED** [review]

**Comment by Maarten-vd-Sande** [comment]
Yes :smile:

**Pull request merged**